src/utils/axios-utils.ts

```diff
@@ -32,23 +32,44 @@ export interface ExtendedAxiosRequestConfig extends AxiosRequestConfig {
  * also removes auth field from the current axios request config.
  *
  * @param {AxiosInstance} axiosInstance - given axios instance
- * @param {Authentication} authentication - authentication object, which will take care of getting
+ * @param {Authentication} authenticationPromise - authentication object, which will take care of getting
  * the authorization header value
  */
 export function setupAuthorizationHeaderOnRequestInterceptor(
     axiosInstance: AxiosInstance,
-    authentication: Authentication
+    authenticationPromise: Promise<any>
```

Comment:
What about setting concrete type?
```suggestion
    authenticationPromise: Promise<Authentication>
```

Reply:
Here we can return a Keystore which does not implement the interface Authentication because it does override the function getAuthorizationHeaderValue. That is why we return any.

Reply:
does not override*